source/common/common/logger.h

```diff
@@ -139,16 +190,33 @@ class Registry {
   /**
    * @return the singleton sink to use for all loggers.
    */
-  static std::shared_ptr<LockingStderrOrFileSink> getSink() {
-    static std::shared_ptr<LockingStderrOrFileSink> sink(new LockingStderrOrFileSink());
+  static DelegatingLogSinkPtr getSink() {
+    static DelegatingLogSinkPtr sink = DelegatingLogSink::init();
     return sink;
   }
 
-  /**
-   * Initialize the logging system from server options.
+  /*
+   * Initialize the logging system with the specified lock and log level.
    */
   static void initialize(uint64_t log_level, const std::string& log_format,
-                         Thread::BasicLockable& lock);
+                         Thread::BasicLockable& lock) {
```

Comment:
I have a minor preference for putting this in the .cc file because nearly every file has to parse this header.